neat_core.c

```diff
@@ -1132,15 +1370,26 @@ neat_flow *neat_new_flow(neat_ctx *mgr)
     if (!rv)
         return NULL;
 
-    rv->fd = -1;
     rv->handle = NULL;
-    rv->writefx = neat_write_via_kernel;
-    rv->readfx = neat_read_via_kernel;
+    rv->writefx = neat_write_to_lower_layer;
+    rv->readfx = neat_read_from_lower_layer;
+#if !defined(USRSCTP_SUPPORT)
+    rv->fd = -1;
```

Comment:
I guess this is the issue with supporting kernel sockets in addition to the usrsctp stack.